### DISCUSSION TITLE
Axios와 XMLHttpRequests

### DISCUSSION THREAD

**Discussion by michoball** [wanted-9th-3team/frontend-study#63] [프론트 기술 질문 공간]
axios를 사용하여 비동기 통신부를 구현하셨는데 axios는 어떤 방식으로 서버와 통신하나요?
XMLHttpRequests 는 무엇인가요?

**Comment by michoball** [comment]
axios는 nodejs나 브라우저를 위한 Promise 기반의 http 클라이언트 입니다.  nodejs 에서는 http 모듈을 사용하고 브라우저에서는 XMLHttpRequests를 사용하여 통신을 구현하게 됩니다. 

XMLHTtpRequest ( XHR ) 는 페이지의 새로고침 없이 서버와 데이터를 주고 받을 수 있게 해주는 api 입니다. XML이라는 이름이 붙었지만 json 방식으로도 데이터를 주고 받을 수 있습니다.  
new 키워드로 객체를 생성해서 open, send 와 같은 메서드를 이용해서 데이터 통신을 구현할 수 있습니다. 

axios는 이 XMLHttpRequests 를 이용한 통신 방식을 보다 쉽게 이용할 수 있도록 해주는 라이브러리 입니다.

**Comment by WooGie911** [comment]
Axios는 node.js와 브라우저를 위한 Promise 기반 HTTP 클라이언트 입니다. 
서버 사이드에서는 네이티브 node.js의 http 모듈을 사용하고, 클라이언트(브라우저)에서는 XMLHttpRequests를 사용합니다.

XMLHttpRequest (XHR) 객체는 서버와 상호작용할 때 사용합니다.
XHR을 사용하면 페이지의 새로고침 없이도 URL에서 데이터를 가져올 수 있습니다. 
이를 활용하면 사용자의 작업을 방해하지 않고 페이지의 일부를 업데이트할 수 있습니다.

또한 XMLHttpRequest는 AJAX 프로그래밍에 많이 사용됩니다.
이름에 XML이 들어가긴 하지만, XMLHttpRequest은 XML 뿐만 아니라 모든 종류의 데이터를 가져올 수 있습니다.
